src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -173,96 +177,147 @@ private static void setLandingPageTitle(String filePath, String pageTitle) throw
-
-            Path repoReportDirectory = Paths.get(outputPath, configToAnalyze.getOutputFolderName());
-            logger.info(
-                    String.format(progressTracker.getProgress() + " "
-                            + MESSAGE_START_ANALYSIS, configToAnalyze.getLocation(), configToAnalyze.getBranch()));
-            try {
-                GitRevParse.assertBranchExists(configToAnalyze, FileUtil.getBareRepoPath(configToAnalyze));
-                GitLsTree.validateFilePaths(configToAnalyze, FileUtil.getBareRepoPath(configToAnalyze));
-                GitClone.cloneFromBareAndUpdateBranch(Paths.get(FileUtil.REPOS_ADDRESS), configToAnalyze);
-
-                FileUtil.createDirectory(repoReportDirectory);
-                generatedFiles.addAll(analyzeRepo(configToAnalyze, repoReportDirectory.toString()));
-            } catch (IOException ioe) {
-                String logMessage = String.format(MESSAGE_ERROR_CREATING_DIRECTORY,
-                        configToAnalyze.getLocation(), configToAnalyze.getBranch());
-                logger.log(Level.WARNING, logMessage, ioe);
-            } catch (GitBranchException gbe) {
-                logger.log(Level.SEVERE, String.format(MESSAGE_BRANCH_DOES_NOT_EXIST,
-                        configToAnalyze.getBranch(), configToAnalyze.getLocation()), gbe);
-                handleAnalysisFailed(configs, configToAnalyze,
-                        String.format(LOG_BRANCH_DOES_NOT_EXIST, configToAnalyze.getBranch()));
-            } catch (InvalidFilePathException ipe) {
-                handleAnalysisFailed(configs, configToAnalyze, LOG_BRANCH_CONTAINS_ILLEGAL_FILE_PATH);
-            } catch (GitCloneException gce) {
-                handleAnalysisFailed(configs, configToAnalyze, LOG_ERROR_CLONING_OR_BRANCHING);
-            } catch (NoAuthorsWithCommitsFoundException nafe) {
-                logger.log(Level.WARNING, String.format(MESSAGE_NO_AUTHORS_WITH_COMMITS_FOUND,
-                        configToAnalyze.getLocation(), configToAnalyze.getBranch()));
-                generatedFiles.addAll(generateEmptyRepoReport(repoReportDirectory.toString(),
-                        Author.NAME_NO_AUTHOR_WITH_COMMITS_FOUND));
-                generateEmptyRepoReport(repoReportDirectory.toString(), Author.NAME_NO_AUTHOR_WITH_COMMITS_FOUND);
-            } catch (Exception e) {
-                StringWriter sw = new StringWriter();
-                e.printStackTrace(new PrintWriter(sw));
-                logger.log(Level.SEVERE, sw.toString());
-                handleAnalysisFailed(configs, configToAnalyze,
-                        String.format(LOG_UNEXPECTED_ERROR, configToAnalyze.getLocation(), sw.toString()));
+        List<AnalysisErrorInfo> analysisErrors = new ArrayList<>();
+        if (cloneSuccessful) {
+            Iterator<RepoConfiguration> itr = configsToAnalyze.iterator();
+            while (itr.hasNext()) {
+                progressTracker.incrementProgress();
```

Comment:
One question: Will this increment be multithread safe?

Reply:
Yes, this should be thread safe. The iterator is operating on the list `configsToAnalyze`. This list is obtained from the `repoLocationMap` for a particular `location`:

```java
List<RepoConfiguration> configsToAnalyze = repoLocationMap.get(location);
```

Since each `location` is handled by a separate thread, logically, we would never have 2 different threads trying to access the same list.